File: Full stack/Domoupravitel/angularapp/src/app/models/person-descriptor.ts
import { PersonType } from "./enums/person-type";
import { Residence } from "./enums/residence";
import { Person } from "./person";
import { Property } from "./property";

export interface PersonDescriptor {
    id: string;
    person: Person;
    property: Property;
    type: PersonType;
    residence: Residence;
    monthsInHouse: number;
    registeredOn?: Date;
    unRegisteredOn?: Date;
}

File: Full stack/Models/PersonDescriptor.cs
using System.ComponentModel.DataAnnotations;
using Models.Enums;

namespace Models
{
    public class PersonDescriptor
    {
        [Required]
        public Guid Id { get; set; }

        [Required]
        public Guid PersonId { get; set; }

        public virtual Person Person { get; set; }

        [Required]
        public Guid PropertyId { get; set; }

        public virtual Property Property { get; set; }

        [Required]
        public PersonType Type { get; set; }

        [Required]
        public Residence Residence { get; set; }

        public int MonthsInHouse { get; set; } = 12;

        public DateTime? RegisteredOn { get; set; }

        public DateTime? UnRegisteredOn { get; set; }
    }
}
